src/embed/dispatch.rs

```diff
@@ -114,14 +116,21 @@ async fn do_open(conns: &mut ConnMap, req: &Request) -> Response {
         return Ok("".into());
     }
 
-    let kv: Box<dyn Store> = match IdbStore::new(&req.db_name[..]).await {
-        Err(e) => {
-            return Err(JsValue::from_str(&format!(
-                "Failed to open \"{}\": {}",
-                req.db_name, e
-            )))
+    let open_req = serde_wasm_bindgen::from_value::<OpenRequest>(req.data.clone())
+        .map_err(|e| JsValue::from_str(&format!("Failed to read open request options: {}", e)))?;
+
+    let kv: Box<dyn Store> = if open_req.use_memstore.is_some() && open_req.use_memstore.unwrap() {
```

Comment:
I think it is possible to somehow do this elegantly with a match pattern:

```
let kv: Box<dyn Store> = match open_req.use_memstore {
  Some(true) => Box::new(MemStore::new()),
  _ => match IDBStore::new(...) ...
}
```

I have not tried this though.